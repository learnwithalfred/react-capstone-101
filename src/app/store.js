import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from '../features/rockets/rocketSlice';
import missionReducer from '../features/missions/missionsSlice';
import themeReducer from '../features/theme/themeSlice';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionReducer,
    theme: themeReducer,
  },
});

export default store;
