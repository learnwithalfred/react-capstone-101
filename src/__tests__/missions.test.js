import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Missions from '../features/missions/missions';
import store from '../app/store';

describe('Missions', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('container should have first child', () => {
    const { container } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(container.firstChild).toBeTruthy();
  });

  it('container should have a table', () => {
    const { container } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(container.querySelector('table')).toBeTruthy();
  });

  it('table should have a header', () => {
    const { container } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(container.querySelector('thead')).toBeTruthy();
  });

  it('table should have a body', () => {
    const { container } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(container.querySelector('tbody')).toBeTruthy();
  });

  it('table should have a row', () => {
    const { container } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(container.querySelector('tr')).toBeTruthy();
  });
});
