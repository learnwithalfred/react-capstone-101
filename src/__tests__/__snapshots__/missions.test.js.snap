// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions should render correctly 1`] = `
<div>
  <div
    class="mt-4 container"
  >
    <table
      class="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Mission Name
          </th>
          <th>
            Description
          </th>
          <th>
            Status
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
